app/notifiers/activity_wrapper.rb

```diff
@@ -0,0 +1,34 @@
+class ActivityWrapper < SimpleDelegator
+  # Send notifications according to input type and recipient
+  #
+  # @param [Object] recipient The recipient of the notification
+  # @param [Symbol] type The type of notification
+  def notify(recipient, type)
+    super(recipient, type).tap do |notification|
+      if notification.email?
```

Comment:
Define a `#notify` method in notifier::base and move all these logics to `#notify`